lib/Backend/IR.cpp

```diff
@@ -373,6 +372,15 @@ Instr::Free()
                     || (stackSym->m_isEncodedConstant && stackSym->constantValue != 0));
             }
         }
+        this->FreeDst();
+    }
+    if (this->GetSrc1())
+    {
+        this->FreeSrc1();
+        if (this->GetSrc2())
+        {
+            this->FreeSrc2();
+        }
     }
```

Comment:
Can there be a case where src1 is not present but src2 is present? either an assert or moving src2 check outside will help?

Reply:
It should not happen normally, but I found instances where we unlink src1 and then call instr->Remove()...  I added to fix this.